leaflet-distance-marker.js

```diff
@@ -31,12 +31,33 @@ L.DistanceMarkers = L.LayerGroup.extend({
 		var cssClass = options.cssClass || 'dist-marker';
 
 		var zoomLayers = {};
-		var length = L.GeometryUtil.length(line);
+		// Get line coords as an array
+        	  if (typeof line.getLatLngs == 'function') {
+                	var coords = line.getLatLngs();
+        	 }
+            	// Get accumulated line lengths as well as overall length
+            	var accumulated = L.GeometryUtil.accumulatedLengths(line);
+            	var length = accumulated.length > 0 ? accumulated[accumulated.length - 1] : 0;
+            	// Position in accumulated line length array
+            	var j = 0;
+            	// Number of distance markers to be added
 		var count = Math.floor(length / offset);
 
 		for (var i = 1; i <= count; ++i) {
 			var distance = offset * i;
-			var position = L.GeometryUtil.interpolateOnLine(map, line, distance / length);
+			// Find the first accumulated distance that is greater
+        		// than the distance of this marker
+                	while (j < length - 1 && accumulated[j] < distance) {
```

Comment:
`length` is the length of the polyline here, not the size of the `accumulated` array.  This could lead to out of bounds problem.  I think it should be `j < accumulated.length - 1` instead.